core/parachain/pvf/pvf_impl.cpp

```diff
@@ -202,64 +216,86 @@ namespace kagome::parachain {
     return true;
   }
 
-  outcome::result<Pvf::Result> PvfImpl::pvfValidate(
-      const PersistedValidationData &data,
-      const ParachainBlock &pov,
-      const CandidateReceipt &receipt,
-      const ParachainRuntime &code_zstd) const {
-    OUTCOME_TRY(pov_encoded, scale::encode(pov));
+  void PvfImpl::pvfValidate(const PersistedValidationData &data,
+                            const ParachainBlock &pov,
+                            const CandidateReceipt &receipt,
+                            const ParachainRuntime &code_zstd,
+                            Cb cb) const {
+    REINVOKE(*pvf_thread_handler_,
+             pvfValidate,
+             data,
+             pov,
+             receipt,
+             code_zstd,
+             std::move(cb));
+    CB_TRY(auto pov_encoded, scale::encode(pov));
```

Comment:
Would it not be easier to just extract the code below to a separate function (maybe a lambda), return the result from it normally and then call the callback with it? Instead of a custom TRY macro.

Reply:
Don't want.